File: module-1/12_Polymorphism/exercise-student/dotnet/BankTellerExercise/SavingsAccount.cs
namespace BankTellerExercise
{
    public class SavingsAccount : BankAccount
    {
        public SavingsAccount(string accountHolder, string accountNumber) : base(accountHolder, accountNumber)
        {

        }

        public SavingsAccount(string accountHolder, string accountNumber, int balance) : base(accountHolder, accountNumber, balance)
        {

        }

        public override int Withdraw(int amountToWithdraw)
        {
            // only perform transaction of positive $ and room for fee
            if (Balance - amountToWithdraw >= 2)
            {
                base.Withdraw(amountToWithdraw);
                // Assess $2 fee if it goes below $150
                if (Balance < 150)
                {
                    base.Withdraw(2);
                }
            }
            return Balance;
        }
    }
}

File: module-1/11_Inheritance/exercise-student/java/src/main/java/com/techelevator/SavingsAccount.java
package com.techelevator;

public class SavingsAccount extends BankAccount {

    public SavingsAccount(String accountHolderName, String accountNumber) {
        super(accountHolderName, accountNumber);

    }

    public SavingsAccount(String accountHolderName, String accountNumber, int balance) {

        super(accountHolderName, accountNumber, balance);

    }

    @Override
    public int withdraw(int amountToWithdraw) {

        if (super.getBalance() > 150 + amountToWithdraw) {
            super.withdraw(amountToWithdraw);

        } else if ((super.getBalance() >= amountToWithdraw)) {
            super.withdraw(amountToWithdraw + 2);

        }

        return getBalance();
    }
}